### DISCUSSION TITLE
Will lights use some kind of disguise block under their texture so you aren't left with holes when adding lamps to your TARDIS?

### DISCUSSION THREAD

**Discussion by MagmaSlime123** [eccentricdevotion/TARDIS#710] [Q&A]
Title, there's no Question template, so I'm asking here. Sorry if this isn't the right place.

Anywho, all-in-one causes lamps to be replaced by display entities, but there's no way of breaking said lamps, unlike the redstone ones we used before due to there bring holes that get left behind upon adding new lamps to a console.

**Comment by eccentricdevotion** [comment]
You can use [Discussions](https://github.com/eccentricdevotion/TARDIS/discussions/categories/q-a) to ask questions, or just ask on Discord, I'll see it there eventually...

You can break TARDIS lights using any pickaxe - however you need to _right_-click them _multiple_ times as they are entities not blocks. If you have the latest resource pack installed you should see the break animation as well, otherwise it will turn to a gravel block.

Blocks drop in survival, but not in creative.

**Comment by eccentricdevotion** [comment]
Not sure what you mean by holes?

**Reply by MagmaSlime123** [reply]
The resourcepack is up-to-date.

Right/left-clicking these, in survival or creative, also does nothing.

![2023-04-27_18 10 06](https://user-images.githubusercontent.com/51277766/235016063-b745b317-0887-41a4-bd81-3022e92015c2.png)

**Comment by eccentricdevotion** [comment]
Lights are not being added properly in builds < 2555. Update and change/repair the TARDIS desktop

**Comment by eccentricdevotion** [comment]
You can use the `/tdev displayitem place light_classic_on` command to test breaking

**Reply by MagmaSlime123** [reply]
Is there a way to kill all current lights, or do I have to fully re-create my TARDIS?

**Reply by eccentricdevotion** [reply]
Select corners of console with TARDIS schematic wand, run `/ts remove` - should replace any LIGHT blocks with REDSTONE_LAMPS

**Reply by eccentricdevotion** [reply]
Can also use `/ts convert [light] [REDSTONE_LAMP|SEA_LANTERN]` to convert those blocks to the TARDIS light of your choice

List of light types can be found here: https://github.com/eccentricdevotion/TARDIS/blob/master/src/main/java/me/eccentric_nz/TARDIS/enumeration/TardisLight.java

**Reply by eccentricdevotion** [reply]
`/ts replace REDSTONE_LAMP [other block]` to change redstone lamps to something else

**Reply by MagmaSlime123** [reply]
Fixed, but they're still un-remove-able unless you use commands. There's no actual lamp block or entity (f3+b).

![2023-04-27_21 03 21](https://user-images.githubusercontent.com/51277766/235044665-b0366ca5-b3ea-4f37-96ac-dc0a3bcf213c.png)

**Comment by MagmaSlime123** [comment]
Resolved via Discord.
